From 756f8ce1289cfe22f7b6f9587f62363878355fd0 Mon Sep 17 00:00:00 2001
From: Kyle Fazzari <git@status.e4ward.com>
Date: Mon, 8 Oct 2012 12:07:02 -0400
Subject: [PATCH] UBUNTU: SAUCE: input: Cypress PS/2 Trackpad fix lost sync
 upon palm contact

Fix "Cypress trackpad gets psmouse lost sync / driver resynced
warnings" by setting the contact count to zero upon palm detection
instead of an invalid value. This accomplishes the same thing upon
palm contact (i.e. nothing) without causing the module to believe
it lost sync.

BugLink: http://bugs.launchpad.net/bugs/1048258

Signed-off-by: Kyle Fazzari <git@status.e4ward.com>
Signed-off-by: Kamal Mostafa <kamal@canonical.com>
Acked-by: Herton Krzesinski <herton.krzesinski@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/cypress_ps2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/mouse/cypress_ps2.c b/drivers/input/mouse/cypress_ps2.c
index c3752be..63b6b2e 100644
--- a/drivers/input/mouse/cypress_ps2.c
+++ b/drivers/input/mouse/cypress_ps2.c
@@ -478,8 +478,8 @@ static int cypress_get_finger_count(unsigned char header_byte)
 				if (finger_count == 2) {
 					finger_count = 5;
 				} else {
-					/* invalid header byte data, set invliad finger count. */
-					finger_count = 6;
+					/* Invalid contact (e.g. palm). Ignore it. */
+					finger_count = 0;
 				}
 			}
 		}
-- 
1.8.0

